src/fromager/candidate.py

```diff
@@ -43,6 +43,20 @@ def __repr__(self) -> str:
             return f"<{self.name}=={self.version}>"
         return f"<{self.name}[{','.join(self.extras)}]=={self.version}>"
 
+    def __hash__(self) -> int:
+        return hash(self.__repr__())
```

Comment:
This is not a good implementation of `__hash__`. The hash and eq methods should operate on the same set of fields. It might be better idea to replace the `Candidate` object with a frozen [dataclass](https://docs.python.org/3/library/dataclasses.html). A frozen dataclass implements eq and hash correctly. You may have to redefine `extras` as `tuple[str, ...] | None` to make it work.

Reply:
I don't think I will need a set in my cache. So won't need to implement this anymore